font/src/ft/mod.rs

```diff
@@ -367,7 +367,13 @@ impl FreeTypeRasterizer {
         };
 
         if face.has_color {
-            Ok(downsample_bitmap(bitmap, face.pixelsize_fixup_factor))
+            let fixup_factor = if face.pixelsize_fixup_factor == 0. {
+                // Fallback
+                self.pixel_size.ceil() as f64 / bitmap.width as f64
+            } else {
```

Comment:
This is the final `pixelsize` value divided by the glyph image width, right? That doesn't look quite right. The pixelsize refers to a font's _nominal height_. This is a) not the same as font's line height (the nominal height is smaller) and b) why the width?

I _think_ that you either want to use the primary font's line height (the cell height, that is) divided by the glyph image height. Or, use the nominal point size configured by the user (the `size`  entry in `alacritty.yml`) divided by the `size` attribute you got from fontconfig for this particular font.

Example:
```
 fc-match -v "Noto Color Emoji:size=12"|grep size
Pattern has 43 elts (size 48)
        size: 104,64(f)(w)
        pixelsize: 109(f)(s)
        pixelsizefixupfactor: 0,114679(f)(w)
```
Where `12 / 104.64 == 0,1146`.

Reply:
In case it helps: https://codeberg.org/dnkl/foot/src/branch/master/font.c#L160

Reply:
>This is the final pixelsize value divided by the glyph image width, right? That doesn't look quite right. The pixelsize refers to a font's nominal height. This is a) not the same as font's line height (the nominal height is smaller) and b) why the width?

@dnkl I'm using the pixelsize thing from a user's config, so I guess I should change bitmap width to pixelsize for a font.

The use of width is because of fixupfactor seems like is for width and not height, so you should calculate height fixup yourself. I haven't found any information about on what I should apply the fixup, however applying it to height doesn't scale things well according to my testing.

Reply:
> I'm using the pixelsize thing from a user's config, so I guess I should change bitmap width to pixelsize for a font.

Isn't the user config in points, not pixels? Anyway, if you want to use the glyph size (and not the emoji font's nominal height), then you should scale using the cell height. But as mentioned, this will lead to too-large-looking emojis. Actually, thinking about it, I guess you could use the nominal height too, but I'm guessing the resulting glyphs will look a bit small?

> The use of width is because of fixupfactor seems like is for width and not height, so you should calculate height fixup yourself. I haven't found any information about on what I should apply the fixup, however applying it to height doesn't scale things well according to my testing.

No, fontconfig uses fonts' **nominal height** when calculating the fixup factor (see example above), and that factor should be applied both to the glyph's height and width (otherwise you'll get disproportional glyphs). In general, fontconfig doesn't care about widths, since most applications render proportional fonts where each glyph has a different width.

Reply:
@dnkl Thanks for clarification, now the fallback returns the exact same value as a fixup.